apis/v1alpha1/generated.proto

```diff
@@ -550,28 +563,49 @@ message HTTPRouteAction {
   optional ConfigMapsDefaultLocalObjectReference extensionRef = 2;
 }
 
-// HTTPRouteFilter defines a filter-like action to be applied to
-// requests.
+// HTTPRouteFilter defines additional processing steps that must be completed
+// during the request or response lifecycle.
+// HTTPRouteFilters are meant as an extension point to express additional
+// processing that must be done in Gateway implementations. Some examples include
+// request or response modification, implementing authentication strategies,
+// rate-limiting, and traffic shaping.
+// API guarantee/conformance is defined based on the type of the filter.
+// TODO(hbagdi): re-render CRDs once controller-tools supports union tags:
+// - https://github.com/kubernetes-sigs/controller-tools/pull/298
+// - https://github.com/kubernetes-sigs/controller-tools/issues/461
+// +union
 message HTTPRouteFilter {
-  // Headers related filters.
-  //
-  // Support: extended
-  // +optional
-  optional HTTPHeaderFilter headers = 1;
+  // Type identifies the filter to execute.
+  // Types are classified into two conformance-levels:
+  // - Core and extended: These filter types and their corresponding configuration
+  //   is defined in this package. All implementations must implement
+  //   the core filters. Implementers are encouraged to support extended filters.
+  //   Definitions for filter-specific configuration for these
+  //   filters is defined in this package.
+  // - Custom: These filters are defined and supported by specific vendors.
+  //   In future, filters showing convergence in behavior across multiple
+  //   implementations will be considered for inclusion in extended or core
+  //   conformance rings. Filter-specific configuration for such filters
+  //   is specified using the ExtensionRef field.
+  // Implementers are encouraged to define custom implementation
+  // types to extend the core API with implementation-specific behavior.
+  //
+  // +unionDiscriminator
+  // +kubebuilder:validation:Required
+  // +required
+  optional string type = 3;
```

Comment:
Is this field still needed after you remove the opaque config map?

Reply:
Yes. There will be multiple core/extended filters and we need a way to discriminate between the types of filters.

Reply:
Not sure that I understand that requirement. I thought that the syntax was that only 1 field may be set, otherwise validation would reject the whole thing?

Reply:
Do you mean to say that since only one field will be set, the set field is what matters and we don't need an additional discriminator (since that's a redundant piece of information)?

Reply:
Yep, that's what I meant :)